src/clarity_vm/clarity.rs

```diff
@@ -626,6 +632,56 @@ impl<'a> ClarityBlockConnection<'a> {
         self.cost_track.unwrap()
     }
 
+    pub fn initialize_epoch_2_1(&mut self) -> Result<(), Error> {
+        let mainnet = self.mainnet;
+        let first_block_height = self.burn_state_db.get_burn_start_height();
+        let pox_prepare_length = self.burn_state_db.get_pox_prepare_length();
+        let pox_reward_cycle_length = self.burn_state_db.get_pox_reward_cycle_length();
+        let pox_rejection_fraction = self.burn_state_db.get_pox_rejection_fraction();
+
+        self.as_transaction(|tx_conn| {
+            tx_conn
+                .with_clarity_db(|db| {
+                    db.set_clarity_epoch_version(StacksEpochId::Epoch21);
+                    Ok(())
+                })
+                .unwrap();
+
+            let pox_2_code = if mainnet {
+                &*BOOT_CODE_POX_MAINNET
+            } else {
+                &*BOOT_CODE_POX_TESTNET
+            };
+
+            let pox_2_contract_id = boot_code_id("pox-2", mainnet);
+            let (ast, _) = tx_conn
+                .analyze_smart_contract(&pox_2_contract_id, pox_2_code)
+                .unwrap();
```

Comment:
Minor nit: might want to replace `.unwrap()` with `.expect()` or even `.map_err().unwrap()` to print out what error happened.

Reply:
Addressed in d6458eccc939ccf7806470012568b90a5470fda9